cpython/Makefile

```diff
@@ -79,11 +97,40 @@ $(ZLIBBUILD)/.patched: $(ZLIBTARBALL)
 	touch $@
 
 
-$(BUILD)/Makefile: $(BUILD)/.patched $(ZLIBBUILD)/.patched
+$(SQLITEBUILD)/libsqlite3.la: $(SQLITETARBALL)
+	[ -d $(ROOT)/build ] || (mkdir $(ROOT)/build)
+	tar -C $(ROOT)/build/ -xf $(SQLITETARBALL)
+	( \
+		cd $(SQLITEBUILD); \
+		emconfigure ./configure; \
+		emmake make; \
+	)
+
+
+$(BZIP2BUILD)/libbz2.a: $(BZIP2TARBALL)
```

Comment:
Seeing this now, I wonder iif `sqlite` and `bzip2` should be part of the `cpython` build (since they are used to statically link into the `cpython` part, rather than used be packages) to declutter the top-level.

Reply:
There is precedence where with `zlib`, too.

Reply:
I've wondered if perhaps we should designates a new "prefix" directory which we set `PREFIX` or `--prefix` to  when we build/install.

Maybe put it outside of `/src` because it doesn't need to be visible to the host and I suspect that having it in a non-shared directory might speed up I/O.

`/usr/local/wasm`?

Reply:
Looking back at your comment, I'm not clear on what you were suggesting. Aren't `sqlite` and `bzip` already in `cpython/Makefile` and tarballs and directories live in `cpython/build`? They do have little bits in the top-level `Makefile` which are unfortunate. I'd like to eliminate the need for that, but haven't figured out how to get it to work without that.

I'm not sure but I think maybe `zlib` is a special case that can get by without that linking in the top-level `Makefile` because `zlib` is bundled into emsdk as an [emscripten port](https://emscripten.org/docs/compiling/Building-Projects.html#emscripten-ports)?

Reply:
> Maybe put it outside of /src because it doesn't need to be visible to the host and I suspect that having it in a non-shared directory might speed up I/O.

We definitely shouldn't ever touch files outside of the source directory.  That will break native (outside of docker) building.  At a minimum, if there's a proven speed advantage for docker, it should be configurable, so we can do something like that in docker but not by default.

> Looking back at your comment, I'm not clear on what you were suggesting. Aren't sqlite and bzip already in cpython/Makefile and tarballs and directories live in cpython/build?

Sorry, you're right.  I was multitasking too much.  These libraries that are dependencies of cpython itself should live where you put them.

> I'm not sure but I think maybe zlib is a special case that can get by without that linking in the top-level Makefile because zlib is bundled into emsdk as an emscripten port?

If there's a way to do this so we do less work on our end, I'm all for it.  I vaguely recall trying and failing at that, but YMMV.

Reply:
emscripten-core/emscripten#8349 🤞